app/src/main/java/com/github/geohunt/app/Challenge.kt

```diff
@@ -0,0 +1,163 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.geohunt.app.model.database.api.Challenge
+import com.github.geohunt.app.model.database.api.Location
+import com.github.geohunt.app.model.database.api.PictureImage
+import com.github.geohunt.app.ui.theme.GeoHuntTheme
+import com.github.geohunt.app.utility.DateFormatUtils
+import java.time.LocalDateTime
+import java.util.concurrent.CompletableFuture
+
+/**
+ * Composable function that displays a challenge given as an argument
+ * Following info is displayed :
+ *      - The thumbnail of the challenge
+ *      - The creator of the challenge
+ *      - The date of the creation of the challenge
+ *      - The time remaining to complete the challenge
+ *      - The pictures submitted by users who already completed the challenge
+ * It also adds a button that allows the user to add a submission to given challenge.
+ * @param challenge The challenge to display
+ */
+@Composable
+fun Challenge(challenge: Challenge) {
+    Column(modifier = Modifier.fillMaxSize()) {
+        Column (horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier.fillMaxWidth()) {
+
+            Spacer(modifier = Modifier.size(20.dp))
+
+            Image(painter = painterResource(id = R.drawable.eiffel),
+                    contentDescription = "Thumbnail of the challenge",
+                    modifier = Modifier
+                            .clip(RoundedCornerShape(10.dp)))
+
+            ClaimButton()
+        }
+
+        ChallengeInformation(challenge)
+
+        Spacer(modifier = Modifier.size(20.dp))
+
+        ClaimPictures(challenge.claims)
+    }
+}
+
+@Composable
+fun ClaimButton() {
+    Button(onClick = { submitPosition() }) {
+        Row (verticalAlignment = Alignment.CenterVertically){
+            Text(text = stringResource(id = R.string.challenge_claim))
+            Icon(painter = painterResource(id = R.drawable.radar_icon),
+                    contentDescription = "Radar icon of claim button")
+        }
+    }
+}
+
+@Composable
+fun ChallengeInformation(challenge: Challenge) {
+    val published = challenge.published
+    val expirationDate = challenge.expirationDate
+
+    Column(modifier = Modifier
+            .fillMaxWidth()
+            .padding(5.dp, 0.dp)) {
+
+        Text(text = stringResource(id = R.string.challenge_created_by, challenge.uid),
+                color = Color.Gray)
+
+        Row(modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween) {
+            GrayText(stringResource(id = R.string.challenge_published, DateFormatUtils.formatDate(published)))
+            GrayText(stringResource(id = R.string.challenge_time_remaining, DateFormatUtils.formatRemainingTime(expirationDate)))
+        }
+    }
+}
+
+@Composable
+fun GrayText(text: String) {
```

Comment:
You use this function to change the text color to grey, but the contrast between the gray color and the white background might not be sufficient for some users, especially those with visual impairments. A higher contrast color should be used instead or the text should be bolded.